src/client/ua_client_connect.c

```diff
@@ -1665,7 +1835,10 @@ UA_Client_disconnectSecureChannel(UA_Client *client) {
 
 UA_StatusCode
 UA_Client_disconnect(UA_Client *client) {
-    UA_LOCK(&client->clientMutex);
+    closeListeningReverseConnectSocket(client);
```

Comment:
This is also called from within disconnectSecureChannel that is called just below.
So shouldn't be necessary here.